react-native-cli/index.js

```diff
@@ -74,6 +75,34 @@ var REACT_NATIVE_PACKAGE_JSON_PATH = function() {
   );
 };
 
+// Use Yarn if available, it's much faster than the npm client.
+// Return the version of yarn installed on the system, null if yarn is not available.
+function yarnVersionIfAvailable() {
```

Comment:
This really asks for being extracted into a separate file.
It is rather complex and has hardcoded versions